lib/utils/sdk.ts

```diff
@@ -47,3 +48,27 @@ export async function getApplication(
 	}
 	return sdk.models.application.get(nameOrSlugOrId, options);
 }
+
+/**
+ * Wraps the sdk organization.getAll method,
+ * restricting to those orgs user is a member of
+ */
+export async function getOwnOrganizations(
+	sdk: BalenaSDK,
+): Promise<Organization[]> {
+	return await sdk.models.organization.getAll({
+		$filter: {
+			organization_membership: {
+				$any: {
+					$alias: 'orm',
+					$expr: {
+						orm: {
+							user: await sdk.auth.getUserId(),
```

Comment:
It might also makes sense to also do an in memory filter based on whether you are an org administrator or not, since just being an org member doesn't give you the rights to create apps.
I'm saying in memory, since `orgnaization_membership_role` is not in oB.

Reply:
No, let's not do that. The CLI must not make these assumptions -- what if we want to also enable a new role that allows app creation in the future? Let's not mix things up -- if the user's role doesn't permit app creation, the request will fail.

Reply:
Right yeah.  We had a discussion about this, and came to the conclusion that it's better to list all the organizations that you're associated with, and then fail with a "You don't have the rights to add an application to this organization" type message, rather than hide those organizations here, since this gives full transparency about what the problem is.  Otherwise users might be confused about why some orgs are not showing.

Reply:
Makes sense. On the other hand people that so far were creating apps normally, might no more be able to do so if they are also members of other orgs and they will now have to select one of the orgs that they are a member of.
So this was just to make this
* non workflow-breaking (in case the CLI is used in CIs)
* also offer a slightly better experience for cloud users (so that you don't get orgs that you are just a member or).

I was thinking something like `organization_membership_role[0].name !== 'member'` or something.

Reply:
> non workflow-breaking (in case the CLI is used in CIs)

Sounds like: Before this PR, a user script around `balena app create` would succeed and create the app in the default org. After this PR, the same script would break because an interactive prompt would be raised for selection between the default org (the only useful selection) and other orgs that would necessarily fail when selected. **In which case, it could be said that this PR, as it is, is both a breaking change and one that adds noise to the workflow.**

We could however counter-argue with:
* "Legit users" (arbitrarily defined as users who are not balena staff testing our system and/or the CLI) are unlikely to script `balena app create`, as they don't create that many apps to warrant scripting. And even for the few legit users who might be doing this, this PR (as it is) would only affect them if their user account was also a member of other orgs. It's unlikely that many legit users would be affected.
* What was referred to above as "adding noise to the workflow" can also be described as adding visibility / to the user's account membership in the platform, which is generally a good thing.
* Users who wish to avoid the interactive prompt can, and should, use the `--organization` flag.

I am not siding with either option (whether or not to filter the orgs), just expanding on what they mean.

Reply:
> Sounds like: Before this PR, a user script around balena app create would succeed and create the app in the default org.

Not always.  The CLI was making a bad assumption before, by passing the username as the org handle, which does not work in all cases for the default org.